master/Makefile

```diff
@@ -70,7 +70,7 @@ build/mock_gen.stamp: $(MOCK_GENERATION_INPUTS)
 	touch $@
 
 .PHONY: check-gen
-check-gen: force-gen gen
+check-gen: force-gen gen build/mock_gen.stamp
```

Comment:
Feels a little surprising to me that ```check-gen``` generates files that ```gen``` doesn't?

I think it is probably fine

Reply:
Yeah seems a little excessive for a ```check-gen-fortest```.

Just throwing out another option master build calls ```build/schema_gen.stamp``` instead of our current ```gen```. I also think the PR as seems fine to me too though.